File: src/structure/java/queue/ArrayQueue.java
package structure.java.queue;

import structure.Queue;
import structure.java.Array;

public class ArrayQueue<E> implements Queue<E> {
    private Array<E> array;
    public ArrayQueue(){
        array = new Array<>();
    }

    public ArrayQueue(int capacity){
        array = new Array<>(capacity);
    }

    @Override
    public void enQueue(E element) {
        array.append(element);
    }

    @Override
    public E deQueue() {
        return array.unshift();
    }

    @Override
    public E getFront() {
        return array.getFirst();
    }

    @Override
    public int getSize() {
        return array.getSize();
    }

    @Override
    public boolean isEmpty() {
        return array.isEmpty();
    }

    @Override
    public String toString() {
        return array.toString();
    }
}

File: src/structure/kotlin/queue/ArrayQueue.kt
package structure.kotlin.queue

import structure.Queue
import structure.java.queue.ArrayQueue
import structure.kotlin.ArrayKt

class ArrayQueue<E> constructor(capacity: Int = 10) : Queue<E> {
    private val array = ArrayKt<E>(capacity)
    override fun enQueue(element: E) {
        array.append(element)
    }

    override fun deQueue(): E = array.unshift()

    override fun getFront(): E = array[0]

    override fun getSize(): Int = array.size

    override fun isEmpty(): Boolean = array.isEmpty

    override fun toString(): String {
        return array.toString()
    }
}

fun main(args: kotlin.Array<String>) {
    val queue = ArrayQueue<Int>()
    for(i in 1..1000000){
        queue.enQueue(i)
    }
    println(queue)
    for (i in 1..999999) queue.deQueue()
    println(queue)

    val queueJv = ArrayQueue<Int>()
    for( i in 1..1000000){
        queueJv.enQueue(i)
    }
    println(queueJv)
    for (i in 1..999999) queueJv.deQueue()
    println(queueJv)
}